### ISSUE TITLE
[1.3] Most basic projects not working in windows 7

### ISSUE THREAD

**Issue by pwnball** [meteor/meteor#6687]
No matter what I try or do. Any project, even a brand new one created through meteor create, it breaks as soon as I navigate to localhost:3000. Meteor 1.2.1 works fine.

```

D:\Development\test>meteor --version
Meteor 1.3

D:\Development\test>meteor
[[[[[ ~\D\Development\test ]]]]]

=> Started proxy.
=> Started MongoDB.

http.js:733
      throw new TypeError('The header content contains invalid characters');
            ^
TypeError: The header content contains invalid characters
    at ClientRequest.OutgoingMessage.setHeader (http.js:733:13)
    at new ClientRequest (http.js:1429:14)
    at Object.exports.request (http.js:1899:10)
    at Array.stream [as 3] (C:\Users\Lightspeed\AppData\Local\.meteor\packages\meteor-tool\1.3.0_3\mt-os.windows.x86_32\dev_b
undle\lib\node_modules\http-proxy\lib\http-proxy\passes\web-incoming.js:108:74)
    at ProxyServer.<anonymous> (C:\Users\Lightspeed\AppData\Local\.meteor\packages\meteor-tool\1.3.0_3\mt-os.windows.x86_32\d
ev_bundle\lib\node_modules\http-proxy\lib\http-proxy\index.js:80:21)
    at Proxy._tryHandleConnections (C:\tools\runners\run-proxy.js:182:20)
    at Server.<anonymous> (C:\tools\runners\run-proxy.js:50:12)
    at Server.emit (events.js:98:17)
    at HTTPParser.parser.onIncoming (http.js:2164:12)
    at HTTPParser.parserOnHeadersComplete [as onHeadersComplete] (http.js:152:23)
    at Socket.socket.ondata (http.js:2022:22)
    at TCP.onread (net.js:528:27)
```

**Comment by trajano** [comment]
There are two things I can think of 

#  `meteor reset` 
# or put it in a path that's shorter, in my case I put it in D:\p\...

**Comment by pwnball** [comment]
@trajano thanks for the reply. 
I just took these steps:

- `meteor create d`
- `cd d:\d`
- `meteor reset`
- 'meteor npm install`
- `meteor`

and ended up with the same problem.

**Comment by pwnball** [comment]
I upgraded to 1.3.1 this morning. Unfortunately the problem remains.

**Comment by trajano** [comment]
I wonder where `http.js` is located.  I can't find the proper file that would have line 733

**Comment by pwnball** [comment]
yeah I was struggling on that 2. I am suspecting its a node file, after some browsing google. I dont have node installed seperate of meteor anymore though :/

**Comment by benjamn** [comment]
Yeah, unfortunately that `http.js` file is baked into the `C:\Users\Lightspeed\AppData\Local\.meteor\packages\meteor-tool\1.3.0_3\mt-os.windows.x86_32\dev_bundle\bin\node` executable file, so you can't set breakpoints in it, though you can see the source code [here](https://github.com/nodejs/node/blob/3123e9a6df4fc500727934f5941cfa1c9e8ba820/lib/http.js#L733).

**Comment by pwnball** [comment]
ah that explains why I'm having some trouble locating the file. @benjamn any suggestions on how I can try to figure out what is causing the problem?

**Comment by trajano** [comment]
I would change the file to actually output the string that had the issue.

   throw new TypeError('The header content contains invalid characters ' + value);

At least you'd know what is invalid.

**Comment by pwnball** [comment]
so.. manually build a manipulated version of nodejs and then store that instead of `C:\Users\Lightspeed\AppData\Local\.meteor\packages\meteor-tool\1.3.0_3\mt-os.windows.x86_32\dev_bundle\bin\node.exe`

I guess that could work. In the meanwhile i reinstalled my entire pc. It was due anyway and hoped it would help. Unfortunately it did not. Fresh windows 7 install, fresh meteor install, same problem.

Ill look into cooking up my own nodejs executable, have not done that before :)

**Comment by abernix** [comment]
@pwnball Have you made any progress on this?  Do you think it's reproducible in the [virtual machines that Microsoft provides](https://developer.microsoft.com/en-us/microsoft-edge/tools/vms/windows/)?

**Comment by pwnball** [comment]
@abernix unfortunately no, but I have been otherwise occupied for a while. I actually started working with meteor again today picking up where I left off. 

I can already confirm that on windows 10 I have the same problem as on windows 7. The downside is, my older project I can keep working on, as long as I dont update my meteor on that repo. 

I can not start a new project due to this problem. I havent got any virtual machines setup at the moment, can do ofcourse, but maybe you have setup some in the meantime to test?

Its about time we nip this sucker in the buttocks :)

**Comment by pwnball** [comment]
So my previous post was after upgrading windows 7 to windows 10 without taking any other actions.

What I did now (so after the upgrade and still encountering the problem)

- Uninstalled meteor
- Searched my computer for meteor folders and deleted the .meteor and paxheader folder that remained with total commander (total commander can handle long paths/filenames)
- Reinstalled meteor
- Ran CMD as admin
- d: -> meteor create foldername
- cd foldername
- meteor

Meteor runs, starts up and when going to localhost:3000, it exists code 8 with the same error as before.

**Comment by pwnball** [comment]
So.. I have setup a virtualbox virtual machine on windows 10, virtualizing a windows 10 pro edition (my pc runs a home edition). 
Long story short: it works flawlessly. The downside is that the virtualbox is so slow I can barely startup my browser to test the site, let alone build the site. This is ofcourse not a meteor issue but a personal one :)

I can't bootup using USB drives, so I am burning the windows 10 ISO later this week and using that to install my pc with it. I guess if that still does not work, it is somehow hardware / driver related..

To be continued!

**Comment by ppotoplyak** [comment]
I am hitting what looks like a similar issue but my context is different.  I had to hack web-incoming.js to provide immediate relief.

Full diff:
```
diff --git a/packages/meteor-tool/.1.3.3.1t63v9e++os.linux.x86_64+web.browser+web.cordova/mt-os.linux.x86_64/dev_bundle/lib/node_modules/http-proxy/lib/http-proxy/passes/web-inc
index 4070eb3..6812371 100644
--- a/packages/meteor-tool/.1.3.3.1t63v9e++os.linux.x86_64+web.browser+web.cordova/mt-os.linux.x86_64/dev_bundle/lib/node_modules/http-proxy/lib/http-proxy/passes/web-incoming.j
+++ b/packages/meteor-tool/.1.3.3.1t63v9e++os.linux.x86_64+web.browser+web.cordova/mt-os.linux.x86_64/dev_bundle/lib/node_modules/http-proxy/lib/http-proxy/passes/web-incoming.j
@@ -104,10 +104,17 @@ web_o = Object.keys(web_o).map(function(pass) {
       if(!options.target) { return res.end(); }
     }
 
-    // Request initalization
-    var proxyReq = (options.target.protocol === 'https:' ? https : http).request(
-      common.setupOutgoing(options.ssl || {}, options, req)
-    );
+    try {
+      // Request initalization
+      var proxyReq = (options.target.protocol === 'https:' ? https : http).request(
+        common.setupOutgoing(options.ssl || {}, options, req)
+      );
+    }
+    catch(err) {
+      console.log(err);
+      // must do an early return as to not take further errors
+      return;
+    }
 
     // Enable developers to modify the proxyReq before headers are sent
     proxyReq.on('socket', function(socket) {
```

See https://github.com/nodejitsu/node-http-proxy/issues/964

There is a trivial repro which brings the Meteor/node process down with a specially crafted HTTP request.  I sent the repro to security@meteor.com.

**Comment by pwnball** [comment]
@ppotoplyak nice work, good to see I am not the only one encountering this and others with more node knowledge bending over it. I have a freshly installed windows 10 pro edition running on my pc now and though Im not sure why, I am not running into this problem anymore! :)

**Comment by abernix** [comment]
I'm going to close this issue because it was never possible to reproduce it and I believe it's isolated to a particular setup or environment.  It's entirely possible that it was due to a Node bug, which if that is the case, I highly recommend updating to a newer version of Meteor which uses a newer version of Node.  Meteor 1.4.1.3 is out now and utilizes Node 4.6.1.

If anyone is still experiencing this problem, it would be good to hear more about the situation: for example, if the problem only happens when you actually navigate to the server an inspection of the headers that your browser is sending could be very helpful.

Without that information though, this is impossible to debug.  Thank you for reporting it originally!

**Comment by usamayousuf13** [comment]
its october 2018 and I am having same issue. Meteor app crashes in production with same error. We are using Meteor 1.3.2.4, i know we should update our meteor version but we have tried and unfortunately our application architecture doesnt allow us to do so. I just want to give a top level architecture design overview of our app, User request from browser lands on Meteor which is then passed to a Node application and then back to browser. I have tried researching alot, but no success.
